rust/lang/security/audit/crypto/math-random.yaml

```diff
@@ -0,0 +1,30 @@
+rules:
+  - id: math-random-used
+    metadata:
+      cwe: 'CWE-338: Use of Cryptographically Weak Pseudo-Random Number Generator (PRNG)'
+      owasp: 'A3: Sensitive Data Exposure'
+      references:
+        - https://cheatsheetseries.owasp.org/cheatsheets/Cryptographic_Storage_Cheat_Sheet.html#secure-random-number-generation
+    message: |
+      The product uses a Pseudo-Random Number Generator (PRNG) in a security context,
+      but the PRNG's algorithm is not cryptographically strong.
+    languages: [rust]
+    severity: WARNING
+
+    pattern-either:
+      - pattern: rand::thread_rng()
+      - pattern: rand::random()
+# issues with pattern inside
+#    - pattern: |
+#        use rand::thread_rng;
+#        ...
+
+#    - pattern-not-inside: |
+#        use rand::CryptoRng;
+#        ...
+#    - pattern-not-inside: |
+#        use rand_core::StdRng;
+#        ...
+#    - pattern-not-inside: |
+#        use rand_core::OsRng;
+#        ...
```

Comment:
The problem here is that you have nested them under the `pattern-either`, this operator works like an `OR` operator. 

Your `pattern-inside` is written correctly and works as expected.
https://semgrep.dev/playground/s/kJPP

Please add testcases for these `pattern-not-inside` cases to your testcode.